.github/workflows/format.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Valve Corporation
+# Copyright 2023 LunarG, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: format
+
+on:
+    push:
+    pull_request:
+        branches:
+            - main
+
+permissions:
+    contents: read
+
+jobs:
+  clang-format:
+    name: clang-format
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        path:
+          - 'include'
+          - 'src'
+          - 'tests'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Run clang-format
+      uses: jidicula/clang-format-action@v4.11.0
```

Comment:
Instead of maintaining our own script it's much simpler to just use this existing action to ensure clang-format is enforced.
https://github.com/jidicula/clang-format-action